devito/dle/transformer.py

```diff
@@ -1,82 +1,56 @@
-                    * 'ntstores': Add pragmas to issue nontemporal stores -- [sT]
-
-                 If ``mode`` is a tuple, the last entry may be used to provide optional
-                 arguments to the DLE transformations. Accepted key-value pairs are: ::
-
-                    * 'blockshape': A tuple representing the shape of a block created
-                                    by loop blocking.
-                    * 'blockinner': By default, loop blocking is not applied to the
-                                    innermost dimension of an Iteration/Expression tree
-                                    to maximize the chances of SIMD vectorization. To
-                                    force the blocking of this loop, the ``blockinner``
-                                    flag should be set to True.
-                    * 'openmp': True to emit OpenMP code, False otherwise.
+    :param mode: Drive the tree transformation. ``mode`` is a string indicating
+                 a certain optimization pipeline. The following values are accepted: ::
+
+                     * 'noop': Do nothing.
+                     * 'basic': Add instructions to avoid denormal numbers and create
+                                elemental functions for rapid JIT-compilation.
+                     * 'advanced': 'basic', vectorization, loop blocking.
+                     * '3D-advanced': Like 'advanced', but apply 3D loop blocking
+                                      if there are at least the perfectly nested
+                                      parallel iteration spaces.
+                     * 'speculative': Apply all of the 'advanced' transformations,
+                                      plus other transformations that might increase
+                                      (or possibly decrease) performance.
+                     * 'yask': Optimize by offloading to the YASK optimizer. Still
+                               work-in-progress; should only be used by developers.
+    :param options: A dictionary with additional information to drive the DLE. The
+                    following values are accepted: ::
+
+                        * 'blockshape': A tuple representing the shape of a block created
+                                        by loop blocking.
+                        * 'blockinner': By default, loop blocking is not applied to the
+                                        innermost dimension of an Iteration/Expression
+                                        tree to maximize vectorization. Set this flag to
+                                        True to override this heuristic.
+                        * 'openmp': True to emit OpenMP code, False otherwise.
     :param compiler: Compiler class used to perform JIT compilation. Useful to
                      introduce compiler-specific vectorization pragmas.
     """
+    assert mode is None or isinstance(mode, str)
```

Comment:
this one should really be an exception, I agree. I'm gonna fix it now, because an assert is really bad ad that point. The first one you pointed out, however, is conceptually more an assert than an exception, I think (anyway, just nitpicking on my side)